wicket-core/src/main/java/org/apache/wicket/protocol/http/request/WebClientInfo.java

```diff
@@ -46,14 +46,20 @@
 
 	/**
 	 * The user agent string from the User-Agent header, app. Theoretically, this might differ from
-	 * {@link org.apache.wicket.protocol.http.ClientProperties#isNavigatorJavaEnabled()} property, which is
-	 * not set until an actual reply from a browser (e.g. using {@link BrowserInfoPage} is set.
+	 * {@link org.apache.wicket.protocol.http.ClientProperties#isNavigatorJavaEnabled()} property,
+	 * which is not set until an actual reply from a browser (e.g. using {@link BrowserInfoPage} is
+	 * set.
 	 */
 	private final String userAgent;
 
 	/** Client properties object. */
 	private final ClientProperties properties;
 
+	private final static UserAgentAnalyzer UAA = UserAgentAnalyzer.newBuilder()
```

Comment:
UserAgentAnalyzer isn't reentrant. Since it is expensive to create one, instances have to be pooled.

Reply:
The initialization is expensive, but the UserAgentAnalyzer is thread safe so it can be reused. Maybe we are able to use MetaDataKey

Reply:
UserAgentAnalyzer#parse(UserAgent) is synchronized, please check its code. This is even mentioned on the project home page.

Reply:
Have a look at the recent changes. A lot of stuff changed already and I think it is ok like it is now (except the tests)